File: LeetCode-Go/03-无重复字符的最长子串.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(lengthOfLongestSubstring("abcabcbb"))
}
func lengthOfLongestSubstring(s string) int {
	sum_str := []byte{}
	i := 0
	j := 0
	max := 0
	lens := len(s)
	for ;i < lens && j < lens; {
		if findIndex(s[j],sum_str) == -1 {
			sum_str = append(sum_str, s[j])
			j ++
		} else {
			i++
			j = i
			sum_str = []byte{}
		}
		if max < len(sum_str){
			max = len(sum_str)
		}
	}
	return  max
}
/**
查询在元素在切片的位置
*/
func findIndex(key byte,s []byte) int{
	for i,value := range s {
		if value == key {
			return i
		}
	}
	return -1
}

File: LeetCode-Python/03-无重复字符的最长子串.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sum_str = ""
        i = k = 0
        lens = len(s)
        max = 0
        while i < lens and k < lens:
            if sum_str.find(s[k]) == -1:
                sum_str += str(s[k])
                k += 1
            else:
                i += 1
                k = i
                sum_str = ""
            if max < len(sum_str):
                max = len(sum_str)

        return  max

